File: TDDKata_Calculator_2011Nov30/Calculator.m
//
//  Created by dgadd on 11-11-30.
//
// To change the template use AppCode | Preferences | File Templates.
//


#import "Calculator.h"


@implementation Calculator

- (BOOL)containsWithin:(NSString *)numbersToAdd searchString:(NSString *)searchString {
    return [numbersToAdd rangeOfString:searchString].location != NSNotFound;
}

- (int)sum:(NSString *)numbersToAdd {
        int total = 0;
        NSArray *numbersArray = [numbersToAdd componentsSeparatedByString:@","];
        for (NSString *numberString in numbersArray)
            total += [numberString intValue];
        return total;
    }

- (void)handleNewLineDelimiter:(NSString **)numbersToAdd {
    (*numbersToAdd) = [*numbersToAdd stringByReplacingOccurrencesOfString:@"\n" withString:@","];
}

- (int)add:(NSString *)numbersToAdd {
    [self handleNewLineDelimiter:&numbersToAdd];
    if ([self containsWithin:numbersToAdd searchString:@","])
        return [self sum:numbersToAdd];

    return [numbersToAdd length] > 0 ? [numbersToAdd intValue] : 0;
}

@end

File: CalculatorApp/Calculator.py
import re

__author__ = 'dgadd'

class Calculator(object):
    def sum(self, numbersToAdd):
        numbersArray = re.split(r',', numbersToAdd)
        total = 0
        for numberString in numbersArray:
            if len(numberString) > 0:
                total += int(numberString)
        return total

    def add(self, numbersToAdd):
        if "," in numbersToAdd:
            return self.sum(numbersToAdd)

        return int(numbersToAdd) if len(numbersToAdd) > 0 else 0